File: 119.Pascal'sTriangleII.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        def factorial(self, n):
            if n < 2:
                return 1
            ans = 1
            for i in range(1, n+1):
                ans *= i
            return ans
        def element(self, n, k):
            if k == 0:
                return 1
            return factorial(self, n) // (factorial(self, k) * factorial(self, n-k))
        return [element(self, rowIndex, i) for i in range(rowIndex+1)]

File: 119.Pascal'sTriangleII.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        if (rowIndex == 0)
            return vector<int>(1, 1);
        vector<int> tab = {1, 1};
        for(int i = 1; i < rowIndex; ++i) {
            vector<int> aux = {1};
            for(int j = 1; j < tab.size(); ++j)
                aux.push_back(tab[j-1] + tab[j]);
            aux.push_back(1);
            tab = aux;
        }
        return tab;
    }
};
